### PR TITLE
Add Mino Plus  Hotswap

### PR DESCRIPTION
**Pull request by CheeseL0ver**
## Description
Add a hotswap version of the Mino Plus PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mino_plus/hotswap/config.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2022 ShandonCodes (@ShandonCodes)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+#define DEBOUNCE 5
+
```
Comment (waffle87) [inline keyboards/mino_plus/hotswap/config.h:6]:
```suggestion
```

File: keyboards/mino_plus/hotswap/hotswap.c [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2022 ShandonCodes (@ShandonCodes)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include "hotswap.h"
```
Comment (waffle87) [inline keyboards/mino_plus/hotswap/hotswap.c:3]:
```suggestion
#include "quantum.h"
```

File: keyboards/mino_plus/hotswap/hotswap.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k2c, k2e,     \
+    k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b, k3c, k3e,  \
+    k40, k41, k42, k45, k49, k4a, k4b, k4c, k4d, k4e\
+) { \
+    { k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b, k0c, k0d, k0e}, \
+    { k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b, k1c, k1d, k1e},  \
+    { k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k2c, KC_NO, k2e}, \
+    { k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b, k3c, KC_NO, k3e}, \
+    { k40, k41, k42, KC_NO, KC_NO, k45, KC_NO, KC_NO, KC_NO, k49, k4a, k4b, k4c, k4d, k4e}, \
+}
```
Comment (waffle87) [inline keyboards/mino_plus/hotswap/hotswap.h:27]: This needs to moved to `info.json`, and this file removed.

File: keyboards/mino_plus/hotswap/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [2] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, QK_BOOT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (waffle87) [inline keyboards/mino_plus/hotswap/keymaps/default/keymap.c:27]:
```suggestion
    )
```

File: keyboards/mino_plus/hotswap/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [3] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, QK_BOOT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (waffle87) [inline keyboards/mino_plus/hotswap/keymaps/via/keymap.c:35]:
```suggestion
    )
```

File: keyboards/mino_plus/mino_plus.h [hunk 6]
```diff
@@ -7,4 +7,8 @@
 
 #ifdef KEYBOARD_mino_plus_soldered
 #    include "soldered.h"
+#endif
```
Comment (waffle87) [inline keyboards/mino_plus/mino_plus.h:10]: This file should be removed.

File: keyboards/mino_plus/hotswap/rules.mk [hunk 7]
```diff
@@ -0,0 +1,3 @@
+OLED_ENABLE = yes
```
Comment (waffle87) [inline keyboards/mino_plus/hotswap/rules.mk:1]: If OLEDs are enabled at the keyboard level, there needs to exist some basic functionality for them at keyboard level as well, in the `hotswap.c` file using the `oled_update_kb()` function.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/mino_plus/hotswap/rules.mk (hunk 7 above)
Comment (CheeseL0ver) [reply keyboards/mino_plus/hotswap/rules.mk:1]: I will move this so it will be enabled at the global level and will leave the OLED code in `mino_plus.c`

**Review by drashna — APPROVED** [review]

**Comment by CheeseL0ver** [comment]
bump: Just wanting to touch base, as it has been a couple weeks with no new activity.

File: keyboards/mino_plus/hotswap/info.json [hunk 8]
```diff
@@ -0,0 +1,473 @@
+{
```
Comment (fauxpark) [inline keyboards/mino_plus/hotswap/info.json:1]: Please run `qmk format-json` on this file. Make sure the branch is up to date with upstream first.

Comment (CheeseL0ver) [reply keyboards/mino_plus/hotswap/info.json:1]: @fauxpark I just updated the file. If there are any issue let me know.

**Review by fauxpark — APPROVED** [review]

**Comment by CheeseL0ver** [comment]
bump: have been approved for a month with no further merge action...

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**